File: api/src/java/org/sakaiproject/yaft/api/YaftPermissions.java
package org.sakaiproject.yaft.api;

public class YaftPermissions
{
	private boolean forumCreate = false;
	private boolean forumDeleteOwn = false;
	private boolean forumDeleteAny = false;
	private boolean discussionCreate = false;
	private boolean discussionDeleteOwn = false;
	private boolean discussionDeleteAny = false;
	private boolean messageCreate = false;
	private boolean messageCensor = false;
	private boolean messageDeleteOwn = false;
	private boolean messageUnDelete = false;
	private boolean messageDeleteAny = false;
	private boolean canModifyPermissions = false;
	private boolean viewInvisible = false;
	private String role = "";
	
	public void setRole(String role)
	{
		this.role = role;
	}
	
	public String getRole()
	{
		return role;
	}

	public void setForumCreate(boolean forumCreate)
	{
		this.forumCreate = forumCreate;
	}

	public boolean isForumCreate()
	{
		return forumCreate;
	}

	public void setDiscussionCreate(boolean discussionCreate)
	{
		this.discussionCreate = discussionCreate;
	}

	public boolean isDiscussionCreate()
	{
		return discussionCreate;
	}

	public void setMessageCreate(boolean messageCreate)
	{
		this.messageCreate = messageCreate;
	}

	public boolean isMessageCreate()
	{
		return messageCreate;
	}

	public void setMessageCensor(boolean messageCensor)
	{
		this.messageCensor = messageCensor;
	}

	public boolean isMessageCensor()
	{
		return messageCensor;
	}

	public void setCanModifyPermissions(boolean canModifyPermissions)
	{
		this.canModifyPermissions = canModifyPermissions;
	}

	public boolean isCanModifyPermissions()
	{
		return canModifyPermissions;
	}

	public void setForumDeleteOwn(boolean forumDeleteOwn)
	{
		this.forumDeleteOwn = forumDeleteOwn;
	}

	public boolean isForumDeleteOwn()
	{
		return forumDeleteOwn;
	}

	public void setForumDeleteAny(boolean forumDeleteAny)
	{
		this.forumDeleteAny = forumDeleteAny;
	}

	public boolean isForumDeleteAny()
	{
		return forumDeleteAny;
	}

	public void setDiscussionDeleteOwn(boolean discussionDeleteOwn)
	{
		this.discussionDeleteOwn = discussionDeleteOwn;
	}

	public boolean isDiscussionDeleteOwn()
	{
		return discussionDeleteOwn;
	}

	public void setDiscussionDeleteAny(boolean discussionDeleteAny)
	{
		this.discussionDeleteAny = discussionDeleteAny;
	}

	public boolean isDiscussionDeleteAny()
	{
		return discussionDeleteAny;
	}

	public void setMessageDeleteOwn(boolean messageDeleteOwn)
	{
		this.messageDeleteOwn = messageDeleteOwn;
	}

	public boolean isMessageDeleteOwn()
	{
		return messageDeleteOwn;
	}

	public void setMessageDeleteAny(boolean messageDeleteAny)
	{
		this.messageDeleteAny = messageDeleteAny;
	}

	public boolean isMessageDeleteAny()
	{
		return messageDeleteAny;
	}

	public void setViewInvisible(boolean viewInvisible)
	{
		this.viewInvisible = viewInvisible;
	}

	public boolean isViewInvisible()
	{
		return viewInvisible;
	}
}

File: tool/src/webapp/js/yaft_permissions.js
function YaftPermissions(data) {

	for(var i=0,j=data.length;i<j;i++) {
		if('yaft.discussion.create' === data[i])
			this.discussionCreate = true;
		else if('yaft.discussion.deleteAny' === data[i])
			this.discussionDeleteAny = true;
		else if('yaft.discussion.deleteOwn' === data[i])
			this.discussionDeleteOwn = true;
		else if('yaft.forum.create' === data[i])
			this.forumCreate = true;
		else if('yaft.forum.deleteAny' === data[i])
			this.forumDeleteAny = true;
		else if('yaft.forum.deleteOwn' === data[i])
			this.forumDeleteOwn = true;
		else if('yaft.message.create' === data[i])
			this.messageCreate = true;
		else if('yaft.message.deleteAny' === data[i])
			this.messageDeleteAny = true;
		else if('yaft.message.deleteOwn' === data[i])
			this.messageDeleteOwn = true;
		else if('yaft.message.read' === data[i])
			this.messageRead = true;
		else if('yaft.modify.permissions' === data[i])
			this.modifyPermissions = true;
		else if('yaft.view.invisible' === data[i])
			this.viewInvisible = true;
	}
}
